### PR TITLE
Fix various bugs related to unspecified call attempts

### PR DESCRIPTION
**Pull request by Horneth**
This also fixes https://broadinstitute.atlassian.net/browse/DSDEEPB-2873

### CODE REVIEW

**Comment by geoffjentry** [comment]
:+1:

**Comment by cjllanwarne** [comment]
Wheel: @kshakir @mcovarr

File: src/main/scala/cromwell/engine/db/slick/ExecutionInfoComponent.scala [hunk 1]
```diff
@@ -36,6 +36,7 @@ trait ExecutionInfoComponent {
     (executionId: Rep[Int], key: Rep[String]) => for {
       executionInfo <- executionInfos
       if executionInfo.key === key
+      if executionInfo.executionId === executionId
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/db/slick/ExecutionInfoComponent.scala]: whoops

File: src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala [hunk 2]
```diff
@@ -77,7 +77,7 @@ object CallMetadataBuilder {
     executionMap => {
       // Remove collector entries for the purposes of this endpoint.
       val inputsNoCollectors = filterCollectorSymbols(callInputs)
-      val inputsByKey = inputsNoCollectors.groupBy(i => ExecutionDatabaseKey(i.scope, i.key.index))
+      val inputsByKey = inputsNoCollectors.groupBy(i => ExecutionDatabaseKey(i.scope, i.key.index, 1))
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala]: so this effectively ignores the attempt in rolling up the inputs?

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 3]
```diff
@@ -261,7 +261,7 @@ class WorkflowManagerActor(backend: Backend) extends LoggingFSM[WorkflowManagerS
   }
 
   private def assertCallExistence(id: WorkflowId, callFqn: FullyQualifiedName): Future[Any] = {
-    globalDataAccess.getExecutionStatus(id, ExecutionDatabaseKey(callFqn, None)) map {
+    globalDataAccess.getExecutionStatus(id, ExecutionDatabaseKey(callFqn, None, 1)) map {
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: shouldn't this take the max attempt?

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala [hunk 4]
```diff
@@ -292,7 +292,7 @@ class WorkflowManagerActor(backend: Backend) extends LoggingFSM[WorkflowManagerS
     for {
       _ <- assertWorkflowExistence(workflowId)
       _ <- assertCallExistence(workflowId, callFqn)
-      outputs <- globalDataAccess.getOutputs(workflowId, ExecutionDatabaseKey(callFqn, None))
+      outputs <- globalDataAccess.getOutputs(workflowId, ExecutionDatabaseKey(callFqn, None, 1))
```
Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: same q here

Comment (mcovarr) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: oh nm, this just validates the call name is valid and 1 is fine for that

**Comment by mcovarr** [comment]
:+1:

File: src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala (hunk 3 above)
Comment (Horneth) [inline src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala]: This method is used to respond to API queries which don't support the concept of attempt (and probably shouldn't). It verifies that a call with this FQN and index exists, which should be true if there was at least one attempt so I think 1 is ok here.

File: src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala (hunk 2 above)
Comment (Horneth) [inline src/main/scala/cromwell/engine/workflow/CallMetadataBuilder.scala]: All attempts have the same inputs so this value doesn't matter I think.

**Pull request merged**